win32ss/printing/base/winspool/printers.c

```diff
@@ -386,8 +386,42 @@ EndPagePrinter(HANDLE hPrinter)
 BOOL WINAPI
 EnumPrintersA(DWORD Flags, PSTR Name, DWORD Level, PBYTE pPrinterEnum, DWORD cbBuf, PDWORD pcbNeeded, PDWORD pcReturned)
 {
+    WCHAR NameW[255];
+    CHAR  NameA[255];
+    BOOL ret;
+    DWORD i;
+    PPRINTER_INFO_4W ppi4w = NULL;
+    PPRINTER_INFO_4A ppi4a = NULL;
+
     TRACE("EnumPrintersA(%lu, %s, %lu, %p, %lu, %p, %p)\n", Flags, Name, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
-    return FALSE;
+    if (Name == NULL)
+    {
+        NameW[0] = UNICODE_NULL;
+    }
+    else
+    {
+    /* https://stackoverflow.com/questions/41147180/why-enumprintersa-and-enumprintersw-request-the-same-amount-of-memory */
+        MultiByteToWideChar(CP_ACP, 0, Name, -1, NameW, ARRAYSIZE(NameW));
+    }
+    ret = EnumPrintersW(Flags, NameW, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
+
+    ppi4w = (PPRINTER_INFO_4W)pPrinterEnum;
+    ppi4a = (PPRINTER_INFO_4A)pPrinterEnum;
+
```

Comment:
ppi4a and ppi4w are overlapping.
Consider input/output buffers.

Reply:
Here the input is ppi4w and the output is ppi4w.
They are overlapping. Does it work fine?

Reply:
Please look at this topic that I referenced in the PR:
https://stackoverflow.com/questions/41147180/why-enumprintersa-and-enumprintersw-request-the-same-amount-of-memory
This points to the fact that these values are most likely converted "in-place" from Unicode to ASCII. So I wanted to be able to point to the Unicode buffer and read from it in Wide characters and then convert to ASCII and write back to the same buffer space in ASCII characters. So the overlapping of these two structures was by intent and this seems to work fine. Thanks.

Reply:
From your link:
```c
    ppi4a = (PPRINTER_INFO_4A)bufa;
    ppi4w = (PPRINTER_INFO_4W)bufw;
```
No, no. Be careful. bufa and bufw are different allocated buffers.

Reply:
pPrinterName is a pointer, and the copying-to is smaller. OK.

Reply:
@katahiromz, you are certainly a more skilled programmer than I am, so I am looking for suggestions on what could be better. My thinking is that my working buffer is pointed to by "pPrinterEnum". Then I create two new pointers to just point to this same buffer. One allows me to use ASCII functions to get to the items within the structure and the other allows me to use Unicode functions to get to these same memory areas. In addition the code works to give me the results that I want. Can you give specific changes that you think are better?